hydromt/components/config.py

```diff
@@ -0,0 +1,215 @@
+            )
+            return
+
+        ext = splitext(p)[-1]
+        if ext in [".yml", ".yaml"]:
+            self._data = read_yaml(read_path)
+        else:
+            raise ValueError(f"Unknown file extention: {ext}")
+
+    @hydromt_step
+    def set(self, data: Dict[str, Any]):
+        """Set the config dictionary at key(s) with values.
+
+        Parameters
+        ----------
+        data: Dict[str,Any]
+            A dictionary with the values to be set. keys can be dotted like in
+            :py:meth:`~hydromt.components.config.ConfigComponent.set_value`
+
+        Examples
+        --------
+        >> self.set({'a': 1, 'b': {'c': {'d': 2}}})
+        >> self.data
+            {'a': 1, 'b': {'c': {'d': 2}}}
+        >> self.set({'a.d.f.g': 1, 'b': {'c': {'d': 2}}})
+        >> self.data
+            {'a': {'d':{'f':{'g': 1}}}, 'b': {'c': {'d': 2}}}
+        """
+        for k, v in data.items():
+            self.update(k, v)
+
+    @hydromt_step
+    def update(self, key: str, value: Any):
+        """Update the config dictionary at key(s) with values.
+
+        Parameters
+        ----------
+        key : str
+            a string with '.' indicating a new level: 'key1.key2' will translate
+            to {"key1":{"key2": value}}
+        value: Any
+            the value to set the config to
+
+        Examples
+        --------
+        >> self.set({'a': 1, 'b': {'c': {'d': 2}}})
+        >> self.data
+            {'a': 1, 'b': {'c': {'d': 2}}}
+        >> self.set_value('a', 99)
+        >> {'a': 99, 'b': {'c': {'d': 2}}}
+
+        >> self.set_value('b.d.e', 24)
+        >> {'a': 99, 'b': {'c': {'d': 24}}}
+        """
+        self._initialize()
+        parts = key.split(".")
+        num_parts = len(parts)
+        current = self.data
+        for i, part in enumerate(parts):
+            if part not in current:
+                current[part] = {}
+            if i < num_parts - 1:
+                current = current[part]
+            else:
+                current[part] = value
+
+    def get_value(self, key: str, abs_path: bool = False):
+        """Get a config value at key(s).
+
+        Parameters
+        ----------
+        args : tuple or string
+            keys can given by multiple args: ('key1', 'key2')
+            or a string with '.' indicating a new level: ('key1.key2')
+        abs_path: bool, optional
+            If True return the absolute path relative to the model root,
+            by deafult False.
+            NOTE: this assumes the config is located in model root!
+
+        Returns
+        -------
+        value : any type
+            dictionary value
+
+        Examples
+        --------
+        >> self.set({'a': 1, 'b': {'c': {'d': 2}}})
+
+        >> get_config_value('a')
+        >> 1
+
+        >> get_config_value('b.c.d')
+        >> 2
+
+        >> get_config_value('b.c')
+        >> {'d': 2}
+
+        """
+        self._initialize()
```

Comment:
No need to initialize, `self.data` does that.

Reply:
good point